onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -982,6 +995,45 @@ TensorrtExecutionProvider::~TensorrtExecutionProvider() {
   ReleaseTensorRTCustomOpDomainList(info_.custom_op_domain_list);
 }
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 10000
+bool TensorrtExecutionProvider::IsGraphCaptureEnabled() const {
+  return cuda_graph_enable_;
+}
+
+bool TensorrtExecutionProvider::IsGraphCaptureAllowed() const {
+  return regular_run_count_before_graph_capture_ >= min_num_runs_before_cuda_graph_capture_;
+}
+
+void TensorrtExecutionProvider::CaptureBegin() {
+  cuda_graph_->Reset();
+  cuda_graph_->CaptureBegin();
+}
+
+void TensorrtExecutionProvider::CaptureEnd() {
+  cuda_graph_->CaptureEnd();
+  is_graph_captured_ = true;
+}
+
+bool TensorrtExecutionProvider::IsGraphCaptured() const {
+  return is_graph_captured_;
+}
+
+Status TensorrtExecutionProvider::ReplayGraph() {
+  ORT_ENFORCE(IsGraphCaptured());
```

Comment:
should this be ORT_ENFORCE(isGraphCaptured) or should we return error status instead if IsGraphCaptured() is false?

Reply:
We did check IsGraphCaptured() before calling ReplayGraph() in [inference_session.cc](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/inference_session.cc#L2120).

Reply:
ok i see. i was just commenting on the fact that ReplayGraph() signature returns a Status. so would it be better to return status rather than do an ORT_ENFORCE assertion which throws exception?

Reply:
I think both approaches work. 
Considering CUDA EP also has ORT_ENFORCE [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cuda/cuda_execution_provider.cc#L203), probably make them consistent at this point.